src/services/statistics.spec.ts

```diff
@@ -1,50 +1,98 @@
 import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
-import { getResultStatistics } from './statistics';
-import * as db from '../db';
 import postgres from 'postgres';
+import * as db from '../db';
 import { createDbPool } from '../utils/db/createDbPool';
 import { runMigrations } from '../utils/db/runMigrations';
+import { insertVotesSchema } from '../types';
 import { cleanup, seed } from '../utils/db/seed';
-import { Request, Response } from 'express';
+import { z } from 'zod';
+import { executeResultQueries } from './statistics';
+import { eq } from 'drizzle-orm';
 
 const DB_CONNECTION_URL = 'postgresql://postgres:secretpassword@localhost:5432';
 
-describe('service: statistics', () => {
+describe('getResultStatistics endpoint', () => {
   let dbPool: PostgresJsDatabase<typeof db>;
-  let req: Partial<Request>;
-  let res: Partial<Response>;
   let dbConnection: postgres.Sql<NonNullable<unknown>>;
+  let userTestData: z.infer<typeof insertVotesSchema>;
+  let otherUserTestData: z.infer<typeof insertVotesSchema>;
+  let cycle: db.Cycle | undefined;
+  let questionOption: db.QuestionOption | undefined;
+  let forumQuestion: db.ForumQuestion | undefined;
+  let user: db.User | undefined;
+  let otherUser: db.User | undefined;
 
   beforeAll(async () => {
     const initDb = createDbPool(DB_CONNECTION_URL, { max: 1 });
     await runMigrations(DB_CONNECTION_URL);
     dbPool = initDb.dbPool;
-
-    // Seed the database
-    await seed(dbPool);
-
-    // Initialize req and res objects
-    req = {};
-    res = {};
-
-    // Initialize dbConnection
     dbConnection = initDb.connection;
+    // seed
+    const { users, questionOptions, forumQuestions, cycles } = await seed(dbPool);
+    // Insert registration fields for the user
+    questionOption = questionOptions[0];
+    forumQuestion = forumQuestions[0];
+    user = users[0];
+    otherUser = users[1];
+    cycle = cycles[0];
+    userTestData = {
+      numOfVotes: 2,
+      optionId: questionOption?.id ?? '',
+      questionId: forumQuestion?.id ?? '',
+      userId: user?.id ?? '',
+    };
+    otherUserTestData = {
+      numOfVotes: 2,
+      optionId: questionOption?.id ?? '',
+      questionId: forumQuestion?.id ?? '',
+      userId: otherUser?.id ?? '',
+    };
+
+    // Add additional data to the Db
+    await dbPool.update(db.cycles).set({ status: 'OPEN' }).where(eq(db.cycles.id, cycle!.id));
```

Comment:
i think the seed now auto makes the cycle open

Reply:
removed